test/hyperbahn-client/discover.js

```diff
@@ -0,0 +1,133 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var DebugLogtron = require('debug-logtron');
+var NullLogtron = require('null-logtron');
+
+var path = require('path');
+var fs = require('fs');
+var crypto = require('crypto');
+
+var HyperbahnClient = require('tchannel/hyperbahn/index.js');
+var TChannelThrift = require('tchannel/as/thrift.js');
+
+var thriftSource = fs.readFileSync(path.join(__dirname, '../../hyperbahn.thrift'), 'utf8');
+
+module.exports = runTests;
+
+if (require.main === module) {
+    runTests(require('../lib/test-cluster.js'));
+}
+
+function runTests(HyperbahnCluster) {
+    HyperbahnCluster.test('discover success', {
+        size: 5
+    }, function t(cluster, assert) {
+        var bob = cluster.remotes.bob;
+        var steve = cluster.remotes.steve;
+
+        var discoverClient;
+
+        var client = new HyperbahnClient({
+            serviceName: 'hello-bob',
+            callerName: 'hello-bob-test',
+            hostPortList: cluster.hostPortList,
+            tchannel: bob.channel,
+            logger: DebugLogtron('hyperbahnClient')
+        });
+
+        client.once('advertised', onAdvertised);
+        client.advertise();
+
+        function onAdvertised() {
+            discoverClient = new HyperbahnClient({
+                serviceName: 'hello-steve',
+                callerName: 'hello-steve-test',
+                hostPortList: cluster.hostPortList,
+                tchannel: steve.channel,
+                logger: DebugLogtron('hyperbahnClient')
+            });
+
+            discoverClient.discover({serviceName: 'hello-bob'}, onDiscovered);
+        }
+
+        function onDiscovered(err, hosts) {
+            assert.error(err, "foo", err);  // Should not have error here
+
+            assert.deepLooseEqual(hosts, [bob.channel.hostPort]);
+
+            client.destroy();
+            discoverClient.destroy();
+            assert.end();
+        }
+    });
+
+    HyperbahnCluster.test('discover no peers', {
+        size: 5
+    }, function t(cluster, assert) {
+        var steve = cluster.remotes.steve;
+
+        var discoverClient = new HyperbahnClient({
+            serviceName: 'hello-steve',
+            callerName: 'hello-steve-test',
+            hostPortList: cluster.hostPortList,
+            tchannel: steve.channel,
+            logger: DebugLogtron('hyperbahnClient')
+        });
+
+        discoverClient.discover(null, onDiscovered);
+
+        function onDiscovered(err, hosts) {
+            assert.error(!err, "NoPeersAvailable response", "expected NoPeersAvailable error");
```

Comment:
For readability please use 

```
assert.ok(err, 'expected error')
assert.equal(err.message, 'NoPeersAvailable reponse');
```

Having two assertions about an error is really useful in tests, it's easy to see a single line referencing the err and thinking "of thats checking there is no error". If you see two assertion lines you know its checking there IS an error.